core/src/main/scala/org/apache/spark/ResourceUtils.scala

```diff
@@ -95,19 +103,12 @@ private[spark] object ResourceUtils extends Logging {
   }
 
   def parseTaskResourceRequirements(sparkConf: SparkConf): Seq[TaskResourceRequirement] = {
-    listResourceIds(sparkConf, SPARK_TASK_PREFIX).map { id =>
-      val settings = sparkConf.getAllWithPrefix(id.confPrefix).toMap
-      val amount = settings.get(AMOUNT).getOrElse(
-        throw new SparkException(s"You must specify an amount for ${id.resourceName}")).toInt
-      TaskResourceRequirement(id.resourceName, amount)
+    parseAllResourceRequests(sparkConf, SPARK_TASK_PREFIX).map { request =>
+      TaskResourceRequirement(request.id.resourceName, request.amount)
     }
   }
 
-  def hasTaskResourceRequirements(sparkConf: SparkConf): Boolean = {
```

Comment:
I deleted this method because `parseTaskResourceRequirements` doesn't cost much.